docs/rfc/block-registration.md

```diff
@@ -0,0 +1,416 @@
+
+See the [Transforms](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional) documentation for more details.
+
+### Deprecated versions
+
+* Type: `object[]`
+* Optional
+* Localized: No
+* Property: `deprecated`
+
+```json
+{ "deprecated": [ {
+	"attributes": {},
+	"save": "my-deprecated-save.js",
+	"supports": {}
+} ] }
+```
+
+This property contains the definition of the deprecated versions of the block type. It is used to ensure that old blocks with old markup are not considered invalid.
+
+See the [Deprecated Blocks](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-deprecation/) documentation for more details.
+
+### Supports
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `supports`
+
+Optional block extended support features. 
+
+See the [block supports](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#supports-optional) documentation page for more details.
+
+### Stylesheets
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `stylesheets`
+
+```json
+{
+	"stylesheets": {
+		"main": "my-block-style.css",
+		"editor": "my-block-editor-style.css",
+		"theme": "my-block-theme-style.css"
+	}
+}
+```
+
+This property is a pointer to CSS files containing the CSS used for the block in different contexts.
+
+## Internationalization
+
+Localized properties are automatically wrapped in `_x` function calls on the back end and the front end of WordPress. These translations are added as an inline script to the `wp-block-library` script handle in WordPress core or to the plugin's script handle when it defines metadata definition.
+
+WordPress string discovery automatically translates these strings using the `textdomain` value specified in the plugin header.
+
+**Example:**
+
+```json
+{
+	"title": "My block",
+	"description": "My block is fantastic",
+	"keywords": [ "fanstastic" ]
+}
+```
+
+In JavaScript, with the help of a Babel plugin, this becomes:
+
+```js
+const metadata = {
+	title: _x( 'My block', 'block title', 'my-plugin' ),
+	description: _x( 'My block is fantastic', 'block description', 'my-plugin' ),
+	keywords: [ _x( 'fanstastic', 'block keywords', 'my-plugin' ) ],
+}
+```
+
+In PHP, it is transformed at runtime to code roughly equivalent to:
+
+```php
+$metadata = array(
+	'title' => _x( 'My block', 'block title', 'my-plugin' ),
+	'description': _x( 'My block is fantastic', 'block description', 'my-plugin' ),
+	'keywords': array( _x( 'fanstastic', 'block keywords', 'my-plugin' ) ),
+);
+```
+
+Implementation should follow the existing [get_plugin_data](https://codex.wordpress.org/Function_Reference/get_plugin_data) function which parses the plugin contents to retrieve the plugin’s metadata, and it applies translations dynamically.
```

Comment:
Thanks, @aduth for all suggestions. I applied all of them. In addition, I included the above sentence as clarification.